src/state/wallet/hooks.ts

```diff
@@ -123,7 +123,9 @@ export function useTokenBalancesWithLoadingIndicator(
     () =>
       account && tokens && tokens.length > 0
         ? tokens.reduce<{ [tokenAddress: string]: TokenAmount | undefined }>((memo, token, i) => {
-            const amount = balances?.[i]?.[0]
+            const amount =
+              token.wrapped.address === balances?.[i]?.[0]?.currency.wrapped.address ? balances?.[i]?.[0] : undefined
```

Comment:
`.toLowerCase()` ?

Reply:
we dont need it because new Token has already parsed the address 
[validateAndParseAddress](https://github.com/KyberNetwork/ks-sdk-core/blob/main/src/utils/validateAndParseAddress.ts#L22)

Reply:
```balances?.[i]?.[0]?.currency.equals(token)```

Reply:
nice

Reply:
`.equals()` might crash invariant. Be careful when use it.

Reply:
<img width="654" alt="image" src="https://github.com/KyberNetwork/kyberswap-interface/assets/11427213/27e17926-753c-4dbd-b8b8-6dfd97f98461">